File: C++/Medium/WrongAnswer/maximum-sum-of-distinct-subarrays-with-length-k_6.cpp
class Solution {
public:
    long long maximumSubarraySum(vector<int>& nums, int k) {
        
        long long res = 0;
        int n = nums.size();
        unordered_map<int, int> m;
        int i = 0;
        long long sum = 0;
        for (int j = 0; j < n; ++j) {
            if (m[nums[j]]++ == 0) sum += nums[j];
            if (j - i + 1 > k) {
                if (--m[nums[i++]] == 0) sum -= nums[i - 1];
            }
            if (j - i + 1 == k) res = max(res, sum);
        }
        return res;
    }
};

File: Java/Medium/WrongAnswer/maximum-sum-of-distinct-subarrays-with-length-k_6.java
class Solution {
    public long maximumSubarraySum(int[] nums, int k) {
        
        int n = nums.length;
        long[] sum = new long[n+1];
        for(int i=0;i<n;i++) sum[i+1] = sum[i] + nums[i];
        long ans = 0;
        TreeSet<Long> set = new TreeSet<>();
        for(int i=k;i<=n;i++){
            set.add(sum[i-k]);
            ans = Math.max(ans, sum[i] - set.first());
        }
        return ans;
    }
}